docker-compose.yml

```
@@ -0,0 +1,52 @@
+version: '3.9'
+services:
+  frontend:
+    build: ./front-end
+    container_name: app_frontend
+    command: npm run start
+    volumes:
+      - ./front-end/src:/frontend/src/
+    ports:
+      - 3000:3000
+    depends_on:
+      backend:
+        condition: service_healthy
+    healthcheck:
+      test: [ "CMD", "lsof", "-t", "-i:3000" ]
+      timeout: 10s
+      retries: 5
+  backend:
+    build: ./back-end
+    container_name: app_backend
+    command: npm run dev
+    volumes:
+      - ./back-end/src:/backend/src/
+    ports:
+      - 3001:3001
+    depends_on:
+      db:
+        condition: service_healthy
+    environment:
+      - PORT=3001
+      - DB_USER=root
+      - DB_PASS=123456
+      - DB_HOST=db
+      - DB_PORT=3306
+    healthcheck:
+      test: [ "CMD", "lsof", "-t", "-i:3001" ]
+      timeout: 10s
+      retries: 5
+  db:
+    image: mysql:8.0.21
+    container_name: db
+    ports:
+      - 3002:3306
+    environment:
+      - MYSQL_ROOT_PASSWORD=123456
+    restart: 'always'
+    healthcheck:
+      test: [ "CMD", "mysqladmin", "ping", "-h", "localhost" ]
```

```suggestion
      test: out=$$(mysqladmin ping -h localhost -P 3306 -u root --password=$$(echo $${MYSQL_ROOT_PASSWORD}) 2>&1); echo $$out | grep 'mysqld is alive' || { echo $$out; exit 1; }
```
Esse teste é mais confiável, mas talvez precise de uns ajustes. Dê uma olhada [nessa thread do StackOverflow](https://stackoverflow.com/a/62986398/4089305)

---

Testei e deu bom, valeuuu ;)